### PR TITLE
Update CMakeLists process translation files in the ISIS source code.

### PR DESCRIPTION
**Pull request by krlberry**
## Description
Updated CMakeLists.txt to 
(1) Create an $ISISROOT/appdata/translations directory
(2) Copy everything in ISIS3/isis/appdata/serialnumbers/*.trn and ISIS3/isis/src/*/apps/*/*.trn 
into it. 

## Related Issue
#3727

## Motivation and Context
App Data into Source Project.

## How Has This Been Tested?
I migrated the tgo CaSSIS translation files locally to test in my clone and they were transferred to the correct locations in $ISISROOT after running a cmake build/install. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by kaitlyndlee — APPROVED** [review]
Looks good!

**Comment by krlberry** [comment]
@jessemapel @Kelvinrr Could one of you take a look at this to see if it seems reasonable? Lacking cmake experience and looking for a review from someone with more cmake experience.

File: isis/CMakeLists.txt [hunk 1]
```diff
@@ -490,6 +497,13 @@ add_custom_target(incs ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
   ${CMAKE_SOURCE_DIR}/src/*/apps/*/*.h ${CMAKE_BINARY_DIR}/inc)
 add_dependencies(isis3 incs)
 
+# Add custom build target to copy modified translation files to the build/appdata/translations
+# directory. (As above - may need to be updated)
+add_custom_target(appdata ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/serialnumbers/*.trn ${CMAKE_SOURCE_DIR}/src/*/apps/*/*.trn
+  ${CMAKE_BINARY_DIR}/appdata/translations)
```
Comment (kaitlyndlee) [inline isis/CMakeLists.txt:504]: Will this copy the base translation files that will be in isis/appdata/translations to the build dir?

Comment (krlberry) [reply isis/CMakeLists.txt:504]: That isn't in this PR and is a very good point! I was planning to expand this to include the other directories (including everything that needs to be migrated from base) in a future PR, unless not having it in here is holding other work up?

Comment (kaitlyndlee) [reply isis/CMakeLists.txt:504]: Oh gotcha. I think it's fine to have it in a PR later with the rest of the base stuff, if everyone else thinks so.

**Review by scsides — APPROVED** [review]
Seems reasonable, would appreciate other opinions

**Pull request merged**